spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinderFactory.java

```diff
@@ -136,11 +140,54 @@ public void destroy() throws Exception {
 			configurationName = name;
 		}
 		Binder<T, ?, ?> binderInstance = getBinderInstance(configurationName);
-		Assert.state(GenericsUtils.getParameterType(binderInstance.getClass(), Binder.class, 0).isAssignableFrom(bindingTargetType),
+		Assert.state(
+				(binderInstance instanceof PollableConsumerBinder
+						&& checkCompatiblePollableBinder(binderInstance, bindingTargetType))
+					|| GenericsUtils.getParameterType(binderInstance.getClass(), Binder.class, 0)
+							.isAssignableFrom(bindingTargetType),
```

Comment:
Ouch. .  may be a private method with a descriptive name? Seems like a lot going on here